Sources/Core/Extensions/String+Extensions.swift

```diff
@@ -18,20 +18,33 @@
 
 import Foundation
 
-// This will eventually be replaced by the rewrite of FBSDKAccessTokenCache
-// for now it is needed as a transient dependency of AccessTokenWallet (via Settings)
+extension String {
+  /**
+   Converts a camelCase string to snake_case
 
-// TODO:
-// This public, canonical type makes this usable from anywhere that knows about access tokens.
-// Internally, it should have a bi-directional codable "mirror type", something like
-// `CacheableAccessToken` that is used for converting a canonical AccessToken to/from a
-// serializable medium. This will help separate the AccessToken type from any specific
-// encoding formats.
-//
-class AccessTokenCache: AccessTokenCaching {
-  var accessToken: AccessToken?
+   - Returns: The string converted to snake case
+   */
+  func snakeCased() -> String? {
+    let pattern = "([a-z0-9])([A-Z])"
+
+    guard let regex = try? NSRegularExpression(pattern: pattern, options: []) else {
+      return self
+    }
+
+    let range = NSRange(location: 0, length: count)
+    return regex.stringByReplacingMatches(in: self, options: [], range: range, withTemplate: "$1_$2").lowercased()
```

Comment:
nit: this is completely optional of course but having multiple parameters on their own lines makes it easier to read and review from smaller devices imo. Don't need to change it, just my thoughts.

Reply:
Will amend